### PR TITLE
window-rules: support title criteria

### PR DESCRIPTION
**Pull request by johanmalm**
Example config:

```
<windowRules>
  <windowRule identifier="foot" title="foo"><action name="Maximize"/></windowRule>
</windowRules>
```

Observe that:

- `foot -T foo` starts maximized
- `xterm -T foo` starts normal

### CODE REVIEW

File: src/window-rules.c [hunk 1]
```diff
@@ -13,20 +14,44 @@
...
+			return false;
+		}
+		return match_glob(rule->identifier, id);
+	} else if (rule->title) {
+		if (!title) {
+			return false;
+		}
+		return match_glob(rule->title, title);
+	} else {
+		wlr_log(WLR_ERROR, "rule has no identifier or title\n");
```
Comment (Consolatis) [inline src/window-rules.c:41]: Another candidate for a `validate()` check in `src/rcxml.c`

File: docs/rc.xml.all [hunk 2]
```diff
@@ -380,7 +380,7 @@
     <windowRule identifier="*">
       <action name="Maximize" />
     </windowRule>
-    <windowRule identifier="some-application" serverDecoration="yes" />
+    <windowRule title="some-title" serverDecoration="yes" />
   </windowRules>
```
Comment (Consolatis) [inline docs/rc.xml.all:384]: Should we add examples for all combinations? E.g
- only identifier (set to `*`, so the first example rule)
- identifier and title set
- only title set

**Comment by Consolatis** [comment]
Only nitpicking. LGTM, didn't test.

**Comment by Consolatis** [comment]
> Example config:
> 
> Observe that:
> 
> * `foot -T foo` starts maximized
> * `xterm -T foo` starts normal

I don't quite understand what this means though.

File: src/window-rules.c (hunk 1 above)
Comment (johanmalm) [reply src/window-rules.c:41]: Have added `validate()` which removes window-rules without criteria.

File: docs/rc.xml.all (hunk 2 above)
Comment (johanmalm) [reply docs/rc.xml.all]: Have added one of each + both.

**Comment by johanmalm** [comment]
> > Example config:
> > Observe that:
> > 
> > * `foot -T foo` starts maximized
> > * `xterm -T foo` starts normal
> 
> I don't quite understand what this means though.

There was a config snippet there which I hadn't indented properly, so it didn't show. Same as the commit message.

**Pull request merged**